### ISSUE TITLE
Inconsistent types in _response_dict. Sometimes list, sometimes string

### ISSUE THREAD

**Issue by owenleej** [czpython/python-amazon-mws#28]
I noticed when I do a list_matching_products _response_dict['ListMatchingProductsResult']['Products']['Product'] is a list if there are multiple search results but a dictionary if there is only 1 search result. The behavior is similar in other places as well. If there are multiple sales rankings, _response_dict['SalesRankings']['SalesRank'] is a list, if there is only one it is a dictionary. I feel like it would be better if these were always lists, like even if there was only one result it would be a list with only one element. Having to catch exceptions if they end up being dictionaries instead of lists is kind of annoying.

**Comment by GriceTurrble** [comment]
Agreed. Controlling these by checking `isinstance` each time is a bit of a pain. Iterating over a list of 1 object would be much more intuitive.

At the same time, this might cause some issues, since the module isn't aware that any given node is about to be a list of identical nodes, and not a dictionary of different node types, particularly if there's only one child node (so it wouldn't know whether it's supposed to be a singleton or a single element in a list).

Not sure how I would handle that one.

**Comment by bloodywing** [comment]
Hi guys: see here. https://github.com/bloodywing/python-amazon-mws/commit/3acf8787557df68cc7706eaf873980d3200447b6 I wrote something like that. You see an example in my commit message.
